src/Client/Psr18.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spawnia\Sailor\Client;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use function Safe\json_encode;
+use Spawnia\Sailor\Client;
+use Spawnia\Sailor\Response;
+
+class Psr18 implements Client
+{
+    protected ClientInterface $client;
+
+    protected string $url;
+
+    protected RequestFactoryInterface $requestFactory;
+
+    protected StreamFactoryInterface $streamFactory;
+
+    public function __construct(
+        ClientInterface $client,
+        string $url = '',
```

Comment:
Is this default useful? I reckon it is more likely to lead to errors and should be mandatory.

Reply:
There's already base url specified in Client implementation sometimes. 

There's the same thing with guzzle, the client can already have base url set so I did not want to enforce it https://github.com/spawnia/sailor/blob/524f8cf2a97b1e105cef92d4253dd951cea13bf3/src/Client/Guzzle.php#L20-L22

`Request` requires `string|UrlInterface` to be passed into a constructor so I thought this is reasonable default.